Adding Piazza support along with miscellaneous changes

nfischer opened the pull request:
### Piazza

This now stores piazza IDs and has an API for accessing/changing these IDs. This should be the first step to having Piazza integration in our project.
### SQL updates

I cleaned up the SQL queries in the python. This should only be a stylistic change. I removed trailing whitespace in the process and saw no errors (so if you see errors on your system, please tell me).
### addSampleData script

This was a little overdue, but the `addSampleData.py` script will set the root list to be node 54 (so we won't need to hardcode 54 in `cdApp.py` once the frontend code relies on the `Root` API instead of the `rootId` field in the "get tree" request). This does not yet change any of the actual code, but just changes our test harness to set the root of the tree in a way similar to how it will be set in the real implementation).

nfischer committed:
Removing trailing white space (seems to not break anything)

nfischer committed:
Adding backend support for saving piazza ID strings

nfischer committed:
Refactoring SQL queries to a common form

nfischer committed:
addSampleData.py now uses the root API to set the root list to contain node 54 for testing

sakekasi commented on backend/cdApp.py:
```diff
@@ -161,15 +170,17 @@ def get(self, course_id):
+            g.db.execute('''UPDATE nodes
+                            SET isroot=1
+                            WHERE id=(?) AND course_id=(?) AND isalive=1''',
+                         [int(root_id), int(course_id)])
```
what if isalive is false? what if no nodes match the criteria?

nfischer replied:
Good call. Right now, it silently fails. I can fix this. Will push in a second.

nfischer committed:
Fixing code to not allow silent errors

nfischer committed:
addSampleData.py will now throw an exception upon error, so this can be used for very minimal API testing

nfischer commented:
Fixed @sakekasi's concerns. I also added exceptions to the `addSampleData.py` script, so if anything fails, it'll cause the entire script to fail, so this can be used for very minimal API testing.
